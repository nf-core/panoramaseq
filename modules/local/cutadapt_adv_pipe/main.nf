// Custom advanced, piped cutadapt for R2 only
process CUTADAPT_ADV_PIPE {
    tag "$meta.id"
    label 'process_medium'
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/17/1758869538eb8e658077cc14cd7a4e76fd9b6d73d3a68f85a70bf292e39e27c5/data' :
        'community.wave.seqera.io/library/cutadapt:5.0--991bbd2e184b7014' }"
    input:
    tuple val(meta), path(reads)
    output:
    tuple val(meta), path("*.trim2.R2.fastq.gz"), emit: reads
    path "versions.yml", emit: versions
    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    cutadapt -m 20 -O 20 -a "polyA=A{20}" -a "QUALITY=G{20}" -n 2 ${reads[1]} \
    | cutadapt  -m 20 -O 3 --nextseq-trim=10  -a "r1adapter=A{18}AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC;min_overlap=3;max_error_rate=0.100000" - \
    | cutadapt -m 20 -O 3 -a "r1polyA=A{18}" - \
    | cutadapt -m 20 -O 20 -g "r1adapter=AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC;min_overlap=20" --discard-trimmed -o ${prefix}.trim2.R2.fastq.gz -
    cat <<-END_VERSIONS > versions.yml
    "CUTADAPT_ADV_PIPE": \$(cutadapt --version)
    END_VERSIONS
    """
}
