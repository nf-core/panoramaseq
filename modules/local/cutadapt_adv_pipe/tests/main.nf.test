nextflow_process {

    name "Test Process CUTADAPT_ADV_PIPE"
    script "../main.nf"
    process "CUTADAPT_ADV_PIPE"
    tag "modules"
    tag "modules_local"
    tag "cutadapt_adv_pipe"

    test("sarscov2 Illumina paired-end [fastq] - R2 only processing") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test', single_end:false ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert process.out.reads != null },
                { assert process.out.reads.get(0).get(1) ==~ ".*\\.trim2\\.R2\\.fastq\\.gz" },
                { assert snapshot(process.out.versions).match("versions_paired_end") },
                { assert snapshot(path(process.out.reads.get(0).get(1)).linesGzip[0]).match() }
            )
        }
    }

    test("sarscov2 cutadapt_adv_pipe different sample") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test2', single_end:false ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert process.out.reads != null },
                { assert process.out.reads.get(0).get(1) ==~ ".*\\.trim2\\.R2\\.fastq\\.gz" },
                { assert snapshot(process.out.versions).match("versions_different_sample") }
            )
        }
    }
}
