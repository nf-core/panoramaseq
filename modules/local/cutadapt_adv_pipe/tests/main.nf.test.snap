{
    "sarscov2 Illumina paired-end [fastq] - R2 only processing": {
        "content": [
            "@ERR5069949.2151832 NS500628:121:HK3MMAFX2:2:21208:10793:15304/2"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T10:38:51.976187824"
    },
    "versions_different_sample": {
        "content": [
            [
                "versions.yml:md5,fe291a6e9607e8456d7d0298f6d79db9"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T10:38:58.419350743"
    },
    "versions_paired_end": {
        "content": [
            [
                "versions.yml:md5,fe291a6e9607e8456d7d0298f6d79db9"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T10:38:51.955928044"
    }
}